accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/exceptions/MissingUnsavedAccessionsException.java

```diff
@@ -19,28 +19,34 @@
 
 import uk.ac.ebi.ampt2d.commons.accession.core.AccessionWrapper;
 
-import java.util.HashMap;
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 public class MissingUnsavedAccessionsException extends RuntimeException {
 
+    private List<AccessionWrapper<?, ?, ?>> missingUnsavedAccessions;
+
     public <MODEL, HASH, ACCESSION> MissingUnsavedAccessionsException(List<AccessionWrapper<MODEL, HASH, ACCESSION>>
-                                                                     unsavedAccessions,
+                                                                              unsavedAccessions,
                                                                       List<AccessionWrapper<MODEL, HASH, ACCESSION>>
-                                                                     retrievedAccessions) {
+                                                                              retrievedAccessions) {
         super("Unsaved objects could not be found: " +
-                generateMessage(unsavedAccessions, retrievedAccessions));
+                generateMissingUnsavedAccessions(unsavedAccessions, retrievedAccessions).toString());
```

Comment:
Or the string generation could be delegated to `getMessage` to avoid calling `generateMissingUnsavedAccessions` twice.